src/wasm.h

```diff
@@ -1672,7 +1672,13 @@ class StructNew : public SpecificExpression<Expression::StructNewId> {
   // case, and binaryen doesn't guarantee roundtripping binaries anyhow.
   ExpressionList operands;
 
-  bool isWithDefault() { return operands.empty(); }
+  bool hasDescriptor() {
+    return type.isStruct() && type.getHeapType().getDescriptorType();
+  }
+
+  bool isWithDefault() {
+    return hasDescriptor() ? (operands.size() == 1) : operands.empty();
+  }
```

Comment:
The comment on line 1670 should be updated?

But, perhaps another option is to add a designated field for this?
```
Expression* descriptor = nullptr;
```

Reply:
Yes, adding an optional designated field probably makes sense.

Reply:
@tlively done, ptal